_posts/2023-03-09-group-equivariant-cnn-1/group-equivariant-cnn-1.Rmd

```diff
@@ -0,0 +1,148 @@
+
+-   By domain, I'm referring to the underlying physical space, and the way it is represented in the input data. For example, images are usually coded as a two-dimensional grid, with values indicating pixel intensities. (At least, that's when the image has a single channel. Otherwise, there are a number of grids, each mapped to a separate channel.)
+-   The task is what we're training the network to do: classification, say, or segmentation. (More accurately, tasks may be different at different stages in the architecture. At each stage, the task in question will have its word to say about how layer design should look.)
+
+For instance, take MNIST. The dataset consists of images of ten digits, 0 to 10, all gray-scale. The task -- unsurprisingly -- is to assign each image the digit represented.
+
+First, consider the domain. A $7$ is a $7$ wherever it appears on the grid. We thus need an operation that is *translation-equivariant*: It flexibly adapts to shifts (translations) in its input. More concretely, in our context, *equivariant* operations are able to detect some object's properties even if that object has been moved, vertically and/or horizontally, to another location. *Convolution*, ubiquitous not just in deep learning, is just such a shift-equivariant operation.
+
+Let me call special attention to the fact that, in equivariance, the essential thing is that "flexible adaptation" (as I casually termed it before). Translation-equivariant operations *do* care about an object's new position; they record a feature not abstractly, but at the object's new position. To see why this is important, consider the network as a whole. When we compose convolutions, we build a hierarchy of feature detectors. That hierarchy should be functional no matter where in the image; in addition, it has to be consistent: Location information needs to be preserved between layers.
+
+Terminology-wise, thus, it is important to distinguish equivariance from *invariance*. An invariant operation, in our context, would still be able to spot a feature wherever it occurs; however, it would happily forget where that feature happened to be. Clearly, then, to build up a hierarchy of features, translation-*invariance* is not enough.
+
+What we've done right now is derive a requirement from the domain, the input grid. What about the task? If, finally, all we're supposed to do is name the digit, now suddenly location does not matter anymore. In other words, once the hierarchy exists, invariance *is* enough. In neural networks, *pooling* is an operation that forgets about (spatial) detail: It only cares about the mean, say, or the maximum value itself. This is what makes it suited to "summing up" information about a region, or a complete image, if at the end we only care about returning a class label.
+
+In a nutshell, we were able to specify design desiderates based on (1) what we're given and (2) what we're tasked with.
```

Comment:
_desiderates_ is a pretty archaic term (in fact, I'm not even sure it's a noun—I think the noun form might be _desiderata_, which is definitely archaic). Requirements? Needs? Whatever you think fits.

Reply:
Yeah ... desiderata, def. not :-)

I thought I'd ascertained, at some time (years ago I'm afraid), that it was a noun ...?
Now looked it up and seems that no ... I might try something like "add to our wishlist" ...